### PR TITLE
fix: wrong block header serialization on avalanche

### PR DESCRIPTION
**Pull request by makramkd**
Due to Avalanche block headers having different contents than regular
Ethereum block headers, go-ethereum does not correctly serialize block
headers into RLP, leading to incorrect arguments to the
storeVerifyHeader function.

See https://docs.avax.network/quickstart/integrate-exchange-with-avalanche#optional-custom-golang-sdk
for details. Quoted:

> If you plan on extracting data from the C-Chain into your own systems
using golang, we recommend using our custom ethclient. The standard
go-ethereum Ethereum client does not compute block hashes correctly
(when you call block.Hash()) because it doesn't take into account the
added ExtDataHash header field in Avalanche C-Chain blocks, which is
used move AVAX between chains (X-Chain and P-Chain).

### CODE REVIEW

**Comment by vreff** [comment]
I see some deleted files in this PR. Are those intentional?

**Comment by makramkd** [comment]
> I see some deleted files in this PR. Are those intentional?

Yup, that's intentional. They're not being used AFAIK. But if you're using them, let me know, and I can put them back.

File: core/scripts/vrfv2/testnet/util.go [hunk 1]
```diff
@@ -159,29 +158,49 @@ func decreasingBlockRange(start, end *big.Int) (ret []*big.Int, err error) {
...
-		)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get header: %+v", err)
-		}
-		rlpHeader, err := rlp.EncodeToBytes(h)
-		if err != nil {
-			return nil, fmt.Errorf("failed to encode rlp: %+v", err)
+		// Avalanche block headers are special, handle them by using the avalanche rpc client
+		// rather than the regular go-ethereum ethclient.
+		if helpers.IsAvaxNetwork(env.ChainID) {
```
Comment (vreff) [inline core/scripts/vrfv2/testnet/util.go:166]: I tried to see about a cleaner way to do this (making an abstractClient interface) but i guess the Headers being different types is tough to get around. LGTM.

Comment (vreff) [reply core/scripts/vrfv2/testnet/util.go:166]: Approved pending tests

**Review by vreff — APPROVED** [review]

File: core/scripts/vrfv2/testnet/util.go (hunk 1 above)
Comment (makramkd) [reply core/scripts/vrfv2/testnet/util.go:166]: Yeah, I tried to do that as well, but pretty much all the types are under a different namespace (e.g `*types.Transaction`, not just `*types.Header`) so this seemed like the only way to get around it for now.

**Pull request merged**